components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5553,57 +5566,152 @@ public void deployAPIRevision(String apiId, String apiRevisionId,
-            // TODO remove this to organization once the microgateway can build gateway based on organization.
-            gatewayManager.deployToGateway(api, organization, environmentsToAdd);
+
+        apiMgtDAO.addAPIRevisionDeployment(apiRevisionUUID, apiRevisionDeployments);
+
+        WorkflowExecutor revisionDeploymentWFExecutor = getWorkflowExecutor(
+                WorkflowConstants.WF_TYPE_AM_REVISION_DEPLOYMENT);
+
+        for (APIRevisionDeployment apiRevisionDeployment : apiRevisionDeployments) {
+            APIRevisionWorkflowDTO revisionWFDto = new APIRevisionWorkflowDTO();
+            try {
+                revisionWFDto.setAPIRevision(apiRevision);
+                revisionWFDto.setExternalWorkflowReference(revisionDeploymentWFExecutor.generateUUID());
+                revisionWFDto.setWorkflowReference(apiRevisionUUID);
+                revisionWFDto.setWorkflowType(WorkflowConstants.WF_TYPE_AM_REVISION_DEPLOYMENT);
+                revisionWFDto.setCallbackUrl(revisionDeploymentWFExecutor.getCallbackURL());
+                revisionWFDto.setStatus(WorkflowStatus.CREATED);
+                revisionWFDto.setTenantDomain(organization);
+                revisionWFDto.setTenantId(tenantId);
+                revisionWFDto.setCreatedTime(System.currentTimeMillis());
+                revisionWFDto.setUserName(userNameWithoutChange);
+                revisionWFDto.setApiName(apiIdentifier.getApiName());
+                revisionWFDto.setApiVersion(apiIdentifier.getVersion());
+                revisionWFDto.setApiProvider(apiIdentifier.getProviderName());
+                revisionWFDto.setEnvironment(apiRevisionDeployment.getDeployment());
+                revisionWFDto.setRevisionId(String.valueOf(revisionId));
+                revisionDeploymentWFExecutor.execute(revisionWFDto);
+                revisionDeploymentWFExecutor.complete(revisionWFDto);
+            } catch (WorkflowException e) {
+                log.error("Unable to execute Revision Deployment Workflow for revision: " + apiRevisionUUID, e);
+                handleException("Unable to execute Revision Deployment Workflow for revision: " + apiRevisionUUID, e);
+            }
+
+            // get the workflow state once the executor is executed.
+            WorkflowDTO wfDTO = apiMgtDAO.retrieveWorkflow(revisionWFDto.getExternalWorkflowReference());
+
+            // only send the notification if approved
+            // wfDTO is null when simple wf executor is used because wf state is not stored in the db.
+            if (wfDTO == null || WorkflowStatus.APPROVED.equals(wfDTO.getStatus())) {
+                apiRevisionDeployment.setDisplayOnDevportal(true);
+                apiMgtDAO.updateAPIRevisionDeployment (apiId,Collections.singleton(apiRevisionDeployment));
+                resumeDeployedAPIRevision(apiId, organization, apiRevisionUUID, String.valueOf(revisionId),
+                        apiRevisionDeployment.getDeployment());
+            }
         }
```

Comment:
Need to send the workflow statues to rest api layer. based on that, need to send the status code (ex: 201 for pending state or 200 for successful state). check changeLifeCycleStatus()

Reply:
As discussed offline checked with the API lifecycle which returns the similar status and the status is not used in frontend.